src/source.rs

```diff
@@ -28,7 +28,15 @@ fn parse_smart_source<T: AsRef<Path>>(block_dev: T) -> Result<Option<u8>> {
         .spawn()
         .context(IoError { path: "(smartctl)".to_owned() })?;
 
-    let root: serde_json::Value = serde_json::from_reader(proc.stdout.unwrap())
+    let result = serde_json::from_reader(proc.stdout.take().unwrap());
+    let status = proc.wait()
+        .context(IoError { path: "(smartctl)".to_owned() })?;
+
+    if !status.success() {
+        return Err(Error::CommandError { command: "smartctl".into(), status });
+    }
```

Comment:
`-n standby` will return exit code 2 if the drive is alseep. 

You can override this with `-n standby,0`, or we can handle exit code 2 in addition to exit code 0. Not sure your preference. 

```suggestion
    match status.code() {
        // smartctl will return status code 2 when a drive is in standby
        Some(0) | Some(2) => {},
        _ => return Err(Error::CommandError { command: "smartctl".into(), status }),
     }
```

I came up with this, because I wanted to try out a match (I wish every language had this syntax!), but perhaps there's a better way :slightly_smiling_face:

Reply:
Looks good to me, thanks!